modules/tinymce/src/themes/silver/main/ts/ui/core/LineHeight.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { Arr, Dimension, Optional, Singleton } from '@ephox/katamari';
+import Editor from 'tinymce/core/api/Editor';
+import { Menu } from 'tinymce/core/api/ui/Ui';
+import * as Settings from '../../api/Settings';
+
+const normaliseLineHeight = (input: string) => Dimension.normalise(input, [ 'length', 'percentage', 'empty' ]).getOr(input);
+
+const getLineHeights = (editor: Editor): Menu.ToggleMenuItemSpec[] => {
+  const options = Settings.getLineHeightFormats(editor);
+
+  const apis = new Map<string, Menu.ToggleMenuItemInstanceApi>();
+  const lastApi = Singleton.destroyable();
+  const callback = () => {
+    const current = normaliseLineHeight(editor.queryCommandValue('LineHeight'));
+    Optional.from(apis.get(current))
+      .fold(
+        () => lastApi.clear(),
+        (api) => {
+          lastApi.set({
+            destroy: () => {
+              api.setActive(false);
+            }
+          });
+          api.setActive(true);
+        }
+      );
+  };
+
+  editor.on('NodeChange', callback);
+
+  return Arr.map(
+    options,
+    (value, i) => ({
+      type: 'togglemenuitem',
+      text: value,
+      onSetup: (api) => {
+        apis.set(normaliseLineHeight(value), api);
```

Comment:
I'm concerned this is never cleaned up on destroy... this seems like it would lead to a potential memory leak. Is there a reason you don't unset the API when the menuitem is destroyed?

- [ ] Consider cleaning up API references in the map on destroy

Reply:
Tested this: the whole map gets cleaned up by the GC, so there's no need to do this.